dnora/trg/write.py

```diff
@@ -66,25 +71,49 @@ def __call__(self, grid: Grid, filename: str, infofilename: str, folder: str) ->
             f.write('$EndMeshFormat\n')
 
             # Write Nodes
+            arr_node = np.stack([
+                grid.nodes(),
+                grid.lon(),
+                grid.lat(),
+                grid.topo(),
+            ]).transpose()
+            fmt_node = '%10.0f%22.8f%22.8f%22.8f'
+
             f.write('$Nodes\n')
             f.write(f"{len(grid.lon()):12.0f}\n")
-            #inds = list(range(1,len(lon)+1))
-            for n in range(len(grid.nodes())):
-                f.write(f"{grid.nodes()[n]:10.0f}{grid.lon()[n]:22.8f}{grid.lat()[n]:22.8f}{abs(grid.topo()[n]):22.8f}\n")
+            np.savetxt(f, arr_node, fmt=fmt_node)
             f.write('$EndNodes\n')
 
             # Write Elements
+            N_bound = len(grid.boundary_inds())
+            arr_bound = np.stack([
+                np.arange(1, N_bound+1),
+                N_bound*[15],
+                N_bound*[2],
+                N_bound*[1],
+                N_bound*[0],
+                grid.boundary_inds()
```

Comment:
While we're at it, there was a +1 in the original code here, but I have no idea whether it should be there as grid.boundary_inds() is already 1-based from what I can see